internal/diagnostics/catalog.go

```diff
@@ -122,35 +114,7 @@ func diagnoseCatalog(catalogDir string, opts CatalogOpts) (group Group) {
 		return
 	}())
 
-	// Diagnose catalog working copy
```

Comment:
Je comprends que `joy diagnose` load maintenant aussi le catalog en pre-run, sauf que s'il y a des problèmes avec le catalog, on ne pourra pas les diagnostiquer ici. Est-ce qu'on ne devrait pas plutôt exclure `joy diagnose` du chargement du catalog en pre-run, pour pouvoir garder ces diagnostiques ici?

Reply:
Both perspectives are valid. I value the insight the diagnostics provide in a valid catalog that isn't behaving as we expect. To me given how integral the catalog is to all things joy, there's no point with other diagnostics if the catalog isn't loadable.

Perhaps there's an argument for aesthetics and personally I don't feel strongly about this.

Reply:
Si joy est brisé sur une machine de développeur parce que le catalogue ne charge pas, j'aimerais qu'ils puissent rouler `joy diagnose` et avoir plus d'informations sur ce qui ne va pas. Et idéalement je ne voudrais pas que `joy diagnose` échoue avec une erreur si le catalogue ne charge pas, ça serait pas très professionnel comme outil de diagnostique. Quelqu'un me dit souvent "consistency is the hobgoblin of little minds", je préfère faire une exception ici pour charger le catalogue explicitement, plutôt que d'être "consistent" mais d'affecter le UX et l'utilité de la commande.